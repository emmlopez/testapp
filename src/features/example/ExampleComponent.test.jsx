import React, { useState as useStateMock } from "react";
import { shallow } from "enzyme";
import ExampleComponent from "./ExampleComponent";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
}));
describe("ExampleComponent tests", () => {
  const setState = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation((init) => [init, setState]);
  });
  it("should render to match snapshot", () => {
    const wrapper = shallow(<ExampleComponent />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should execute onclick function for increment button", () => {
    const wrapper = shallow(<ExampleComponent />);
    const button = wrapper.findWhere(
      (node) => node.props().htmlId === "increment"
    );
    expect(button).toHaveLength(1);
    button.simulate("click");
    expect(setState).toHaveBeenCalled();
  });
  it("should execute onclick function for decrement button", () => {

    const wrapper = shallow(<ExampleComponent/>);
    const button = wrapper.findWhere(
      (node) => node.props().htmlId === "decrement"
    );
    expect(button).toHaveLength(1);
    button.simulate("click");
    expect(setState).toHaveBeenCalled();
  });
});
