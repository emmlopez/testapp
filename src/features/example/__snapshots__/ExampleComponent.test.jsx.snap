// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent tests should render to match snapshot 1`] = `
<div>
  <div>
    Click Count: 0
  </div>
  <div>
    <div
      className="button-container"
    >
      <Button
        htmlId="increment"
        onClick={[Function]}
      >
        Increment
      </Button>
      <Button
        htmlId="decrement"
        onClick={[Function]}
      >
        Decrement
      </Button>
    </div>
  </div>
</div>
`;
